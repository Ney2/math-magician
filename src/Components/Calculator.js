/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import calculate from '../Logic/claculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      operation: '',
      next: '',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (e) => {
    this.setState((value) => calculate(value, e.target.innerText));
  }

  render() {
    const { total, operation, next } = this.state;
    return (
      <main>
        <section className="container">
          <div className="display"><p className="output">{(total || '') + (operation || '') + (next || '') || 0 }</p></div>
          <button type="button" className="clear" onClick={this.handleClick()}>AC</button>
          <button type="button" className="plusminus" onClick={this.handleClick()}>+/-</button>
          <button type="button" className="percentage" onClick={this.handleClick()}>%</button>
          <button type="button" className="division" onClick={this.handleClick()}>&divide;</button>
          <button type="button" className="seven" onClick={this.handleClick()}>7</button>
          <button type="button" className="eight" onClick={this.handleClick()}>8</button>
          <button type="button" className="nine" onClick={this.handleClick()}>9</button>
          <button type="button" className="multi" onClick={this.handleClick()}>&times;</button>
          <button type="button" className="four" onClick={this.handleClick()}>4</button>
          <button type="button" className="five" onClick={this.handleClick()}>5</button>
          <button type="button" className="six" onClick={this.handleClick()}>6</button>
          <button type="button" className="minus" onClick={this.handleClick()}>-</button>
          <button type="button" className="one" onClick={this.handleClick()}>1</button>
          <button type="button" className="two" onClick={this.handleClick()}>2</button>
          <button type="button" className="three" onClick={this.handleClick()}>3</button>
          <button type="button" className="plus" onClick={this.handleClick()}>+</button>
          <button type="button" className="zero" onClick={this.handleClick()}>0</button>
          <button type="button" className="point" onClick={this.handleClick()}>.</button>
          <button type="button" className="equal" onClick={this.handleClick()}>=</button>
        </section>
      </main>
    );
  }
}

Calculator.defaultProps = { total: 0, operation: '', next: 0 };
export default Calculator;
