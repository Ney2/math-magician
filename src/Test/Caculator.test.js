import React from 'react';
import Renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

it('renders correctly', () => {
    const tree = Renderer
        .create(<Calculator />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});