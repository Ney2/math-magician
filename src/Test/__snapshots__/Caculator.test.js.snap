// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="wrapper"
>
  <h1>
    Let\`s do some Math!
  </h1>
  <div
    className="container"
  >
    <p
      className="output"
    >
      0
    </p>
    <button
      className="clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="plusminus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="percentage"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="division"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="multi"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="minus"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="point"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</main>
`;
